src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+        user.setLogin("SuperLongLogin");
+        user.setPassword("short");
+        user.setAge(20);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_young_noOk() {
+        User user = new User();
+        user.setLogin("SuperLogin");
+        user.setPassword("LongPassword");
+        user.setAge(17);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_and_login_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("LongPassword");
+        user.setAge(3);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_and_password_notOk() {
+        User user = new User();
+        user.setLogin("LongLogin");
+        user.setPassword("abc");
+        user.setAge(5);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_login_noNull_noOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("someLongPassword");
+        user.setAge(56);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_password_noNull_noOk() {
+        User user = new User();
+        user.setLogin("SomeFancyLogin");
+        user.setPassword(null);
+        user.setAge(56);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_noNull_noOk() {
+        User user = new User();
+        user.setLogin("SuperExoticLogin");
+        user.setPassword("SuperExoticPassword");
+        user.setAge(null);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_login_has_whitespace_noOk() {
+        User user = new User();
+        user.setLogin("SomeLogin have whitespace");
+        user.setPassword("SuperExoticPassword");
+        user.setAge(63);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_password_has_whitespace_noOk() {
+        User user = new User();
+        user.setLogin("SomeLoginNoWhiteSpace");
+        user.setPassword("Super ExoticPassword");
+        user.setAge(63);
+
+        assertThrows(InvalidUserDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    public void user_age_below_0_noOk() {
```

Comment:
The test method name `user_age_below_0_noOk` should be more descriptive. Consider renaming it to `user_age_negative_notOk` to clearly indicate the specific validation being tested.